### ISSUE TITLE
Update concatenation dimension in tensorflow implementation of mutator to reflect image_data_format setting in keras.json config

### ISSUE THREAD

**Issue by code-fury** [microsoft/nni#2945]
**What would you like to be added**:

The concatenation axis in `src/sdk/pynni/nni/nas/tensorflow/mutator.py` is fixed to 0 which does not reflect image_data_format setting in keras.json config. 

**Why is this needed**:

This will cause the model to concatenate multiple inputs in the skip connections in the wrong dimensions. Users might get `Incompatible shapes` error if he uses multiple filter sizes.

**Without this feature, how does current nni work**：
nni only allows concatenating in the first dimension. This will fail if `image_data_format` is `channel_last`.

**Components that may involve changes**:
 src/sdk/pynni/nni/nas/tensorflow/mutator.py

**Brief description of your proposal if any**:
```python
        if reduction_type == 'concat':
            image_data_format = tf.keras.backend.image_data_format()
            if image_data_format == "channels_first":
                axis = 0
            else:
                axis = -1
            return tf.concat(tensor_list, axis=axis)
```

**Comment by code-fury** [comment]
I created a PR for this here: https://github.com/microsoft/nni/pull/2946

**Comment by liuzhe-lz** [comment]
Patch included in release v1.9.
